.github/workflows/nightly-release.yaml

```diff
@@ -0,0 +1,616 @@
+        required: false
+        default: false
+        type: boolean
+      test_bucket:
+        description: 'Test bucket for fork releases (defaults to auto-generated)'
+        required: false
+        default: ''
+        type: string
+      production_bucket:
+        description: 'Production bucket for main releases'
+        required: false
+        default: 'gs://tekton-releases-nightly/pipeline'
+        type: string
+  # uncomment the following to enable manual testing of the workflow
+  # push:
+  #   branches:
+  #     - test-rel
+  #   paths:
+  #     - '.github/workflows/nightly-release.yaml'
+  #     - 'tekton/**'
+  #     - 'cmd/**'
+  #     - 'pkg/**'
+
+env:
+  REGISTRY: ghcr.io
+  REPOSITORY: ${{ github.repository }}
+  KUBERNETES_VERSION: ${{ inputs.kubernetes_version || 'v1.33.0' }}
+  DRY_RUN: ${{ inputs.dry_run || false }}
+  FORCE_RELEASE: ${{ inputs.force_release || false }}
+  TEST_BUCKET: ${{ inputs.test_bucket || '' }} # This applies only while testing in a fork
+  PRODUCTION_BUCKET: ${{ inputs.production_bucket || 'gs://tekton-releases-nightly/pipeline' }}
+
+jobs:
+  pre-checks:
+    name: Pre-release Checks
+    runs-on: ubuntu-latest
+    outputs:
+      should_release: ${{ steps.changes.outputs.should_release }}
+      version_tag: ${{ steps.version.outputs.version_tag }}
+      latest_sha: ${{ steps.version.outputs.latest_sha }}
+      is_fork: ${{ steps.repo_check.outputs.is_fork }}
+      ko_extra_args: ${{ steps.repo_check.outputs.ko_extra_args }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 50
+
+      - name: Repository analysis
+        id: repo_check
+        run: |
+          set -euo pipefail
+
+          echo "Analyzing repository configuration..."
+
+          if [[ "${{ github.repository }}" == "tektoncd/pipeline" ]]; then
+            echo "PRODUCTION MODE: tektoncd/pipeline (upstream)"
+            echo "is_fork=false" >> "$GITHUB_OUTPUT"
+            echo "ko_extra_args=--preserve-import-paths" >> "$GITHUB_OUTPUT"
+          else
+            echo "FORK MODE: ${{ github.repository }}"
+            echo "is_fork=true" >> "$GITHUB_OUTPUT"
+            echo "ko_extra_args=" >> "$GITHUB_OUTPUT"
+          fi
+
+          echo "Repository: ${{ github.repository }}"
+          echo "Owner: ${{ github.repository_owner }}"
+
+      - name: Get latest commit and version
+        id: version
+        run: |
+          set -euo pipefail
+
+          if [[ "${{ github.repository }}" == "tektoncd/pipeline" ]]; then
+            # For upstream, get the latest commit from main
+            REPO="https://github.com/tektoncd/pipeline.git"
+            echo "Fetching latest commit from upstream main branch..."
+          else
+            # For forks, use the current repository
+            REPO="https://github.com/${{ github.repository }}.git"
+            echo "Fetching latest commit from fork main branch..."
+          fi
```

Comment:
Could you clarify why we need this?
All the logic to support running this on forks complicates the codebase and was not really in scope for the migration of nightly builds to GHA.